packages/core/src/utils.js

```diff
@@ -655,18 +654,7 @@ export function retrieveSchema(schema, rootSchema = {}, formData = {}) {
     return {};
   }
   let resolvedSchema = resolveSchema(schema, rootSchema, formData);
-  if ("allOf" in schema) {
-    try {
-      resolvedSchema = mergeAllOf({
-        ...resolvedSchema,
-        allOf: resolvedSchema.allOf,
-      });
-    } catch (e) {
-      console.warn("could not merge subschemas in allOf:\n" + e);
-      const { allOf, ...resolvedSchemaWithoutAllOf } = resolvedSchema;
-      return resolvedSchemaWithoutAllOf;
-    }
-  }
+
```

Comment:
Why remove the allOf logic?

Reply:
@epicfaace  I have refactored allof and it now works more in line with https://json-schema.org/understanding-json-schema/reference/combining.html. before my changes the playground example does nothing, please take a look at updated playground example.

I still have to implement allof with if as per https://json-schema.org/understanding-json-schema/reference/conditionals.html?highlight=else.

another reasoning behind this removal is that current merge for allof does not work with if statement and giving it a resolver will not allow tie back of if / then

I aim to get allof working with if's to essentially do a case statement as I need fairly soon for the product which I am using this lib for.

thanks for your prompt response, hopefully we can get this nailed correctly in a prompt manner,



Likewise I still need to update the tests with regards to the allof stuff.

Reply:
> before my changes the playground example does nothing, please take a look at updated playground example.

Thanks for pointing that out. The old allOf example was not the right example, but the allOf feature did work. I've added a working allOf example in this PR: https://github.com/rjsf-team/react-jsonschema-form/pull/1623/files#diff-4c77c9a609ff92a2b66c66edb321f45a